docs/Release_Management.md

```diff
@@ -0,0 +1,50 @@
+# Release Management
+
+## Overview
+This document describes **Kubernetes Secrets Store CSI Driver** project release management, which talks about versioning, branching and cadence.
+
+## Legend
+
+- **X.Y.Z** refers to the version (git tag) of Secrets Store CSI Driver that is released. This is the version of the Secrets Store CSI Driver image.
+ 
+- **Milestone** should be designed to include feature sets to accommodate monthly release cycles including test gates. GitHub milestones are used by maintainers to manage each release. PRs and Issues for each release should be created as part of a corresponding milestone.
+
+- **Test gates** should include soak tests and upgrade tests from the last minor version.
+
+
+## Versioning
+This project strictly follows [semantic versioning](https://semver.org/spec/v2.0.0.html). All releases will be of the form _vX.Y.Z_ where X is the major version, Y is the minor version and Z is the patch version.
+
+### Patch releases
+- Patch releases provide users with bug fixes and security fixes. They do not contain new features.
+
+### Minor releases
+- Minor releases contain security and bug fixes as well as _**new features**_.
+
+- They are backwards compatible.
+
+### Major releases
+- Major releases contain breaking changes. Breaking changes refer to schema changes, flag changes, and behavior changes of Secrets Store CSI Driver that may require a clean install during upgrade and it may introduce changes that could break backward compatibility.
```

Comment:
For flag changes, do we need a major release? Following along the Kubernetes release pattern, if we want to deprecate an existing flag, we could do that in a minor release itself after n+2 releases?

I agree on schema and/or behavior changes would warrant a major release because that could technically change the way users will need to configure.

Reply:
Make sense. I'll update the doc